index.js

```diff
@@ -50,10 +50,10 @@ var _outputCache = {
 
     helpers: {
 
-        setHeadersOnCacheItem: function onSetHeaders(req, res, cacheItem, headers) {
+        setHeadersOnCacheItem: function onSetHeaders(req, res, cacheItem) {
 
-            cacheItem.headers = headers;
-            var responseCacheHeader = headers['cache-control'];
+            cacheItem.headers = JSON.parse(JSON.stringify(res._headers));
```

Comment:
Ok great! I'm going to merge but have also done a complete rewrite of the module, which is to be released soon. I will incorporate your changes.